src/components/RecordTypesSelect/components/ChooseExistingRecord.js

```diff
@@ -0,0 +1,168 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import classNames from 'classnames';
+
+import {
+  Row,
+  Col,
+  Headline,
+} from '@folio/stripes/components';
+
+import { Triangle } from './Triangle';
+import { RecordItem } from './RecordItem';
+import { RecordSelect } from './RecordSelect';
+import { getElement } from '../../LineBetween/utils';
+
+import css from '../RecordTypesSelect.css';
+
+const defaultSpacing = 50;
+const maxCompareElemWidth = 100;
+const maxHintElemWidth = 180;
+
+export const ChooseExistingRecord = ({
+  incomingRecord,
+  existingRecord,
+  setIncomingRecord,
+  setExistingRecord,
+}) => {
+  const [top, setTop] = useState(null);
+  const [height, setHeight] = useState(null);
+  const [compareElemWidth, setCompareElemWidth] = useState();
+  const [hintElemWidth, setHintElemWidth] = useState();
+
+  /** @type {React.RefObject<HTMLElement>} */
+  const compareElemRef = useRef();
+  /** @type {React.RefObject<HTMLElement>} */
+  const hintElemRef = useRef();
+  /** @type {React.RefObject<HTMLElement>} */
+  const existingRecordsRef = useRef();
+
+  const spaceForCompareElem = Math.min(compareElemWidth / 2 + 4, maxCompareElemWidth) || defaultSpacing;
+
+  useEffect(() => {
+    setCompareElemWidth(Math.min(compareElemRef.current.getBoundingClientRect().width, maxCompareElemWidth));
+    setHintElemWidth(Math.min(hintElemRef.current.getBoundingClientRect().width, maxHintElemWidth));
+  }, []);
+
+  useEffect(() => {
+    const selectedExistingRecordElem = getElement(`#${existingRecord.type}`, existingRecordsRef.current);
+
+    if (selectedExistingRecordElem) {
+      setTop(selectedExistingRecordElem.offsetTop);
+      setHeight(selectedExistingRecordElem.getBoundingClientRect().height);
+    }
+  }, [existingRecord, compareElemWidth, hintElemWidth]);
+
+  const render = typeof top === 'number';
```

Comment:
Use `isNumber` from `lodash` since we have it.
```suggestion
  const render = isNumber(top);
```